lib/elixir/lib/code.ex

```diff
@@ -1386,6 +1387,11 @@ defmodule Code do
     * `:warnings_as_errors` - causes compilation to fail when warnings are
       generated. Defaults to `false`.
 
+    * `:undefined_variable_as_call` (since v1.15.0) - when `true`, undefined
+      variables will emit a warning and be expanded as a local call to the zero-arity
+      function of the same name (`node` would be expanded as `node()`), when `false`
+      will just trigger a compile error. Defaults to `false`.
```

Comment:
Note: The [original proposal ](https://github.com/elixir-lang/elixir/pull/12272#issuecomment-1328226599) was to add the `on_undefined_variable: :raise | :warn` option.

However, I noticed that all existing compiler options were either booleans or lists, so I thought it might be better to keep the consistency especially since this option can only have two values. WDYT?

Reply:
Let's go the boolean route because we may choose to expose other values in the future, such as `:apply` that we already have.

Reply:
> we may choose to expose other values in the future, such as :apply that we already have.

I want to confirm my understanding: `:apply` is a possible value of the `:if_undefined` meta, but there doesn't seem to be a `prematch=apply` as of today? (this would be `:warn` which expands undefined variables, but not silently.

The currently available `prematch` values are:

```elixir
{Read, Counter} | warn | raise | pin | {bitsize,PreVars,OriginalVars}
```

I think only `warn` and `raise` make sense as a compiler default value, the other ones all being context sensitive. Unless we add `apply` from now, but it would be un-natural as we are moving to delete it?

> Let's go the boolean route

I assume from the context you mean "Let's not go"? 🙂
I'm fine with either the boolean or the atoms, I just wanted to confirm the point above first.

Reply:
Yes, let’s not go. :) I don’t think we will support apply but it was an example that the behavior may grow beyond Boolean. :)

Reply:
Got it, thanks for confirming!